File: backend/app/services/opportunity_status.py
from __future__ import annotations


def opportunity_status_label(net_edge: float | None, tolerance: float = 0.001) -> str:
    if net_edge is None or abs(net_edge) <= tolerance:
        return "Near fair value"
    if net_edge > 0:
        return "Possible YES underpricing"
    return "Possible YES overpricing"

File: frontend/lib/opportunityStatus.ts
export type OpportunityStatus = {
  label: "Possible YES underpricing" | "Possible YES overpricing" | "Near fair value";
  tone: "positive" | "negative" | "neutral";
};

export function opportunityStatus(netEdge: number | null | undefined, tolerance = 0.001): OpportunityStatus {
  if (netEdge == null || Math.abs(netEdge) <= tolerance) {
    return { label: "Near fair value", tone: "neutral" };
  }
  if (netEdge > 0) {
    return { label: "Possible YES underpricing", tone: "positive" };
  }
  return { label: "Possible YES overpricing", tone: "negative" };
}
